### PR TITLE
trace memory of MemTables

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

Ref #205 

Added a new method to flush internal metrics, including memory used by `MemTables`.

### CODE REVIEW

**Comment by tabokie** [comment]
I ran a benchmark to get a baseline memory usage:
```
# ./target/release/stress --path /data3/re_bench --time 60 --regions 1000 --write-threads 1 --purge-interval 0 --compact-count 0 --write-without-sync
Memory usage: 8003MiB
```

File: src/memtable.rs [hunk 1]
```diff
@@ -902,6 +908,16 @@ impl MemTableAccessor {
         }
     }
 
+    pub(crate) fn flush_metrics(&self) {
```
Comment (Connor1996) [inline src/memtable.rs:911]: let the user to call `flush_metrics` instead of doing itself?

File: src/engine.rs [hunk 2]
```diff
@@ -92,6 +98,19 @@ where
             listeners.clone(),
         );
 
+        let (tx, rx) = mpsc::channel();
+        let stats_clone = stats.clone();
+        let memtables_clone = memtables.clone();
+        let metrics_flusher = ThreadBuilder::new()
+            .name("raft-engine-metrics".into())
+            .spawn(move || loop {
```
Comment (Connor1996) [inline src/engine.rs:106]: let the user to call flush_metrics instead of doing itself?

Comment (tabokie) [reply src/engine.rs:106]: Okay.

**Pull request merged**